### ISSUE TITLE
Add drgn.Symbol.binding and drgn.Symbol.kind

### ISSUE THREAD

**Issue by osandov** [osandov/drgn#109]
As discussed in #108, the drgn.Symbol API currently provides no way to distinguish between global and local symbols, or symbols for a variable vs symbols for a function. ELF symbols do provide this information as the `ST_BIND` and `ST_TYPE` fields in `Elf_Sym::st_info`: https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-79797.html. We should export these fields in drgn.Symbol (and the underlying `struct drgn_symbol` in libdrgn).

drgn currently only supports ELF, but I try to not leak ELF implementation details into the drgn API. So rather than directly using the ELF definitions of `STB_*` and `STT_*`, we should add our own enum definitions that correspond to the ELF definitions but could also be used for other formats in the future.

So, for symbol binding, we want something like this in libdrgn:

```c
// In libdrgn/drgn.h.in
enum drgn_symbol_binding {
	DRGN_SYMBOL_BINDING_UNKNOWN,
	DRGN_SYMBOL_BINDING_LOCAL,
	DRGN_SYMBOL_BINDING_GLOBAL,
	DRGN_SYMBOL_BINDING_WEAK,
};

enum drgn_symbol_binding drgn_symbol_binding(struct drgn_symbol *sym);

// In libdrgn/symbol.h
struct drgn_symbol {
	...
	enum drgn_symbol_binding binding;
};
```

This corresponds to `STB_LOCAL`, `STB_GLOBAL`, and `STB_WEAK` from ELF, but we also want an unknown value in case we don't recognize the ELF value (e.g., because it's a platform-specific binding type or because the file is corrupted).

We also want a corresponding [Python enum](https://docs.python.org/3/library/enum.html) and attribute:

```python
class SymbolBinding(enum.Enum):
    LOCAL = ...
    GLOBAL = ...
    WEAK = ...

class Symbol:
    ...
    binding: SymbolBinding
```

The enum should be generated from the libdrgn definition by `libdrgn/build-aux/gen_constants.py`. The attribute is just some boilerplate in `libdrgn/python/symbol.c`.

It's the same idea for the symbol type, except we should probably call it `drgn_symbol_kind`/`drgn.SymbolKind` instead of symbol type to avoid confusion with programming language types (I'm not 100% set on that, so please let me know if you disagree).

A couple of miscellanea:
* We probably want a helper function to initialize a `struct drgn_symbol` from an ELF symbol instead of manually constructing it in multiple places (`drgn_program_find_symbol_by_address_internal`, `find_symbol_by_name_cb`). Something like `void drgn_symbol_from_elf(const char *name, uint64_t address, GElf_Sym *sym, struct drgn_symbol *ret)`. That's where you would translate `STB_*`/`STT_*` -> `enum drgn_symbol_binding`/`enum drgn_symbol_kind`.
* `drgn_symbol_eq` needs to compare the new fields.

**Comment by osandov** [comment]
@brenns10 offered to help with this. Please let me know if you have any questions.

**Comment by osandov** [comment]
Implemented by @brenns10 in #115.
